pkg/test/collector_intermediate_test.go

```diff
@@ -143,6 +143,8 @@ func TestCollectorToIntermediate(t *testing.T) {
 		defer conn.Close()
 		conn.Write(templatePacket)
 		conn.Write(dataPacket1)
+		// sleep to ensure dataPacket1 is processed first so that the total count fields are using values from dataPacket1
+		time.Sleep(time.Millisecond)
```

Comment:
This is TCP connection right. Ord
er of packets should be followed. Did you get a chance to run the test with count=500 or some large number?